### DISCUSSION TITLE
Technology Integrations with StarRocks

### DISCUSSION THREAD

**Discussion by alberttwong** [StarRocks/starrocks#26601] [How-tos, Tutorials, Best Practices and Architecture Articles]
![StarRocks Technology Partnership Logos](https://github.com/StarRocks/starrocks/assets/749093/eb5a724b-a3ad-43cd-aad9-5ee5b782ef01)

https://stackshare.io/celerdata/starrocks-technology-integrations

Apache Flink: https://docs.starrocks.io/en-us/latest/loading/Flink-connector-starrocks#overview
Apache Kafka: https://docs.starrocks.io/en-us/latest/loading/Kafka-connector-starrocks
Airbyte: https://github.com/StarRocks/starrocks/discussions/23713
Red Hat OpenShift: https://github.com/StarRocks/starrocks/discussions/22767
DBT: https://github.com/StarRocks/starrocks/tree/main/contrib/dbt-connector
Tableau: https://github.com/StarRocks/starrocks/discussions/23211
Apache Superset: https://github.com/StarRocks/starrocks/discussions/23210
Metabase: https://github.com/StarRocks/starrocks/discussions/23209
Docker: https://github.com/StarRocks/starrocks-kubernetes-operator
Apache Iceberg and Tabular.io: https://github.com/StarRocks/starrocks/discussions/23427, https://iceberg.apache.org/vendors/
Apache Hudi and Onehouse.ai: https://github.com/StarRocks/starrocks/discussions/22947
Microsoft Excel: https://github.com/StarRocks/starrocks/discussions/23628
Apache Hive: https://docs.starrocks.io/en-us/latest/data_source/catalog/hive_catalog#hive-metastore
Delta Lake: https://docs.starrocks.io/en-us/latest/data_source/catalog/deltalake_catalog
JSON: https://github.com/StarRocks/starrocks/discussions/25504
Parquet: https://github.com/StarRocks/starrocks/discussions/22777
Kubernetes: https://docs.starrocks.io/en-us/latest/deployment/sr_operator#deploy-starrocks-operator
Minio: https://docs.starrocks.io/en-us/latest/loading/cloud_storage_load#load-data-from-an-s3-compatible-storage-system
Cloud Canal: https://www.cloudcanalx.com/us/
MindsDB: https://docs.mindsdb.com/data-integrations/starrocks
LangChain: https://python.langchain.com/docs/integrations/providers/starrocks, https://python.langchain.com/docs/integrations/vectorstores/starrocks
Apache Inlong: https://inlong.apache.org/docs/data_node/load_node/starrocks/
Apache Paimon: https://docs.starrocks.io/en-us/latest/data_source/catalog/paimon_catalog
Datadog: https://docs.starrocks.io/en-us/latest/integrations/other_integrations/datadog_support, https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/doc/integration/integration-with-datadog.md
Prometheus + Grafana: https://docs.starrocks.io/en-us/latest/administration/Monitor_and_Alert, https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/doc/integration/integration-prometheus-grafana.md
Apache Dolphin Scheduler: https://dolphinscheduler.apache.org/en-us
Apache SeaTunnel: https://seatunnel.apache.org/docs/connector-v2/sink/StarRocks
Apache Linkis: https://linkis.apache.org/docs/latest/user-guide/datasource-manual/#313-jdbc-engineconn-supported-by-linkis
Apache Ranger: https://github.com/StarRocks/starrocks/pull/29154
Rising Wave: https://docs.risingwave.com/docs/current/sink-to-starrocks/

In addition, there is more content on integrations with StarRocks - https://docs.starrocks.io/en-us/latest/integrations/other_integrations/intro#data-integration

In addition, there may be known issues with the integrations.  Please search github issues for the latest info.

**Comment by kkmsun** [comment]
What framework do you use for these integrations?

**Reply by alberttwong** [reply]
it depends.  what integration are you referring to?

**Comment by kkmsun** [comment]
Connectors to various systems..

**Reply by alberttwong** [reply]
Which one specifically.  Can you can imagine, there are different integrations for different product types.

**Comment by kkmsun** [comment]
I get it, that was the question, if there is a connection framework you are
using or have built.

Thx k

**Comment by svdimchenko** [comment]
Hi there! It would be nice to add terraform provider implementation to be able to manage following resourses:
- catalogs
- databases
- roles
- users
- groups
- grants

See an implementation example for [redshift](https://github.com/brainly/terraform-provider-redshift/tree/master)

**Reply by alberttwong** [reply]
The commerical version of StarRocks, CelerData has a terraform provider https://registry.terraform.io/providers/CelerData/celerdatabyoc/latest/docs.   However if you want to build one for community, we welcome to the contribution.

**Reply by svdimchenko** [reply]
@alberttwong thanks for sharing info about existing terraform provider. However is seems that even in commercial version there is no support for mentioned resources management. So adding them can give your product +1 advantage over other competitors.

**Reply by alberttwong** [reply]
Did you create the github issue to track this request?

**Reply by svdimchenko** [reply]
yeah, created the one: https://github.com/StarRocks/starrocks/issues/33184, it's actually the 3rd one since yesterday when I started my evaluation 😄

**Comment by ourcoteam** [comment]
Lookerstudio ?

**Reply by alberttwong** [reply]
Have you tried the mysql driver?
